mlflow/tensorflow/_autolog.py

```diff
@@ -52,11 +59,162 @@ def on_epoch_end(self, epoch, logs=None):
+    Input Dataset's tensors must contain tuple of (features, labels) that are
+    used for tensorflow/keras train or fit methods
+
+
+    :param dataset: a tensorflow batched/unbatched dataset representing tuple of (features, labels)
+    :return: a numpy array of length `INPUT_EXAMPLE_SAMPLE_ROWS`
+
+    Examples
+    --------
+    >>> input_dataset = tensorflow.data.Dataset.from_tensor_slices(
+    ...                 ({'SepalLength': np.array(list(range(0, 20))),
+    ...                 'SepalWidth': np.array(list(range(0, 20))),
+    ...                 'PetalLength': np.array(list(range(0, 20))),
+    ...                 'PetalWidth': np.array(list(range(0, 20)))},
+    ...                 np.array(list(range(0, 20))))).batch(10)
+    >>> _extract_input_example_from_batched_tf_dataset(input_dataset)
+    {'SepalLength': array([0, 1, 2, 3, 4]),
+    'SepalWidth': array([0, 1, 2, 3, 4]),
+    'PetalLength': array([0, 1, 2, 3, 4]),
+    'PetalWidth': array([0, 1, 2, 3, 4])}
+
+    """
+    limited_df_iter = list(dataset.take(INPUT_EXAMPLE_SAMPLE_ROWS))
+    first_batch = limited_df_iter[0]
+    if isinstance(first_batch, tuple):
+        features = first_batch[0]
+        if isinstance(features, dict):
+            return _extract_sample_numpy_dict(features)
+        elif isinstance(features, (np.ndarray, tensorflow.Tensor)):
+            return _extract_input_example_from_tensor_or_ndarray(features)
+        raise TypeError(f"Unsupported type for features of type: {type(features)}")
+    raise TypeError(f"Unsupported type for dataset batch slice of type: {type(first_batch)}")
+
+
+def extract_data_from_tf_input_fn(input_fn):
+    """
+    Extracts sample data from dict (str -> ndarray),
+    ``tensorflow.Tensor`` or ``tensorflow.data.Dataset`` type.
+
+    :param input_fn: Tensorflow's input function used for train method
+    :return: a slice (of limit ``mlflow.utils.autologging_utils.INPUT_EXAMPLE_SAMPLE_ROWS``)
+             of the input of type `np.ndarray`
+    """
+
+    input_training_data = input_fn()
+    input_features = None
+    if isinstance(input_training_data, tuple):
+        features = input_training_data[0]
+        if isinstance(features, dict):
+            input_features = _extract_sample_numpy_dict(features)
+        elif isinstance(features, (np.ndarray, tensorflow.Tensor)):
+            input_features = _extract_input_example_from_tensor_or_ndarray(features)
+    elif isinstance(input_training_data, tensorflow.data.Dataset):
+        input_features = _extract_input_example_from_batched_tf_dataset(input_training_data)
+    return input_features
+
+
+def get_tf_keras_input_data_slice(input_training_data):
+    """
+    Generates a sample ndarray or dict (str -> ndarray)
+    from the input type 'x' for keras ``fit`` or ``fit_generator``
+
+    :param input_training_data: Keras input function used for ``fit`` or
+                                ``fit_generator`` methods
+    :return: a slice of type ndarray or
```

Comment:
Following from https://github.com/mlflow/mlflow/pull/5510/files#r852379015, can we update this to also say that we return `None` if the contents of `input_training_data` are unsupported?

Reply:
@bali0019 Can we also add a simple test case for unsupported inputs to make sure that we return `None` and don't throw?